engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -68,9 +68,10 @@ def get_instance_id(params)
         nil,
         params['instance_data']
       )
-      instance_id_key = INSTANCE_ID_KEYS[params['hwinfo']['cloud_provider'].downcase.to_sym]
-      iid = verification_provider.parse_instance_data
-      iid[instance_id_key]
+      csp = params['hwinfo']['cloud_provider'].downcase
+      instance_id_key = INSTANCE_ID_KEYS[csp.to_sym]
+      instance_data = verification_provider.parse_instance_data
+      csp.casecmp('microsoft').zero? ? instance_data['attestedData'][instance_id_key] : instance_data[instance_id_key]
```

Comment:
Could this be encapsulated as a method in the cloud provider specific instance verification class?

Reply:
I think we should keep CSP specific cases out of the generic code. If we need to pass a key that is basically meaningless to SCC but used to allow activation for certain products then that key should be provided by `verification_provider.get_identifier`. With this the generic RMT code does the same thing for every provider and the provider specific implementation about how to get this identifier is hidden. We choose to use the instance identifier, a 3rd party CSP in their implementation of the verification code could choose some other identifier.

Reply:
good point